crates/subspace-core-primitives/src/objects.rs

```diff
@@ -95,21 +95,32 @@ pub enum PieceObject {
         /// Object hash
         #[cfg_attr(feature = "serde", serde(with = "hex"))]
         hash: Blake3Hash,
-        // TODO: This is a raw record offset, not a regular one
-        /// Offset of the object in that piece
+        /// Regular record offset of the object in that piece
         offset: u32,
     },
 }
 
 impl PieceObject {
+    /// Converts a raw piece offset to a regular piece offset.
+    #[inline]
+    pub const fn raw_offset_to_regular(raw_offset: u32) -> u32 {
+        raw_offset + raw_offset / Scalar::SAFE_BYTES as u32
+    }
+
+    /// Converts a regular piece offset to a raw piece offset.
+    #[inline]
+    pub const fn regular_offset_to_raw(regular_offset: u32) -> u32 {
+        regular_offset - regular_offset / Scalar::FULL_BYTES as u32
+    }
```

Comment:
I do not have a strong opinion, but it feels that these methods don't really belong here. If I had an offset, I probably wouldn't reach out here looking for conversion method.

Knowing what you know now, was maybe raw offset was the right thing to store all along if you need to convert it for retrieval purposes anyway?